install/install.go

```diff
@@ -338,9 +347,53 @@ func validateConfiguration() error {
 	return nil
 }
 
-func generateCertificate() {
+func generateCertificate(org string) (cert bytes.Buffer, key bytes.Buffer, err error) {
+
 	log.Infof("Generating certificate/key pair - private key in %s", data.key)
-	//TODO: Generate certificate if -g=true is specified
+
+	priv, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	notBefore := time.Now()
+	notAfter := notBefore.Add(5 * 365 * 24 * time.Hour) // 5 years
+
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	template := x509.Certificate{
+		SerialNumber: serialNumber,
+		Subject: pkix.Name{
+			Organization: []string{org},
+		},
+		NotBefore:             notBefore,
+		NotAfter:              notAfter,
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		BasicConstraintsValid: true,
+	}
+
+	derBytes, err := x509.CreateCertificate(rand.Reader, &template, &template, &priv.PublicKey, priv)
+	if err != nil {
```

Comment:
I'm not familiar with Go lang certification, so please make sure it's self signed.